flight/Modules/Sensors/sensors.c

```diff
@@ -291,46 +287,38 @@ static void SensorsTask(void *parameters)
 #endif
 		}
 
-		queue = PIOS_SENSORS_GetQueue(PIOS_SENSOR_BARO);
-		if (queue != NULL) {
-			if (PIOS_Queue_Receive(queue, &baro, 0) != false) {
-				// we can use the timeval because it contains the current time stamp (PIOS_DELAY_GetRaw())
-				last_baro_update_time = timeval;
-				update_baro(&baro);
-				AlarmsClear(SYSTEMALARMS_ALARM_TEMPBARO);
-			} else {
-				// Check that we got valid sensor datas
-				uint32_t dT_baro_datas = PIOS_DELAY_DiffuS(last_baro_update_time);
-				// if the last valid sensor datas older than 100 ms report an error
-				if (dT_baro_datas > MAX_TIME_BETWEEN_VALID_BARO_DATAS_MS) {
-					AlarmsSet(SYSTEMALARMS_ALARM_TEMPBARO, SYSTEMALARMS_ALARM_ERROR);
-				}
-			}
-
+		if (PIOS_SENSORS_GetData(PIOS_SENSOR_BARO, &baro, 0) != false) {
```

Comment:
This change lights up the baro error on the system health gadget, if the board doesn't have one. Calling IsRegistered() to see whether one was initialized via board init seems needed (old code does that by checking whether there's a queue initialized).

Reply:
Are you sure?  Seems that if the receive fails, we do the same GetMissing check the old code did when queue was null.

Reply:
Yea, system health reports a baro error on my Revo clone without baro.

Looking at the code again, GetMissing will return false, so it'll drop to the last else section, which will set an error alarm, since the time-delta will grow past the threshold, because last_baro_update_time won't get updated a couple of lines above, due to a lack of baro readouts.

Would probably still make sense to wrap the whole section in a registration check.